src/lib/Select.svelte

```diff
@@ -454,76 +634,38 @@
 
     function handleFocus(e) {
         if (focused && input === document?.activeElement) return;
-        if (e) dispatch('focus', e);
+        if (e) {
+            onfocus(e);
+        }
         input?.focus();
         focused = true;
     }
 
     async function handleBlur(e) {
         if (isScrolling) return;
         if (listOpen || focused) {
-            dispatch('blur', e);
+            onblur(e);
             closeList();
             focused = false;
             activeValue = undefined;
             input?.blur();
         }
     }
 
-    function handleClick() {
+    function handleClick(ev) {
+        ev.preventDefault();
         if (disabled) return;
-        if (filterText.length > 0) return listOpen = true;
+        if (filterText && filterText.length > 0) return (listOpen = true);
```

Comment:
```suggestion
        if (filterText?.length ?? 0 > 0) return (listOpen = true);
```

As it is, while the result is unchanged, incorrectly filters out the empty string value, to which one *can* query for its length.